aidb/inference/bound_inference_service.py

```diff
@@ -189,63 +201,21 @@ async def infer(self, inputs: pd.DataFrame):
           results.append(df)
         else:
           inference_results = self.service.infer_one(inp_row)
-
-          if len(inference_results) > 0:
-            # FIXME: figure out where to put the column renaming
-            for idx, col in enumerate(self.binding.output_columns):
-              if idx < len(inference_results.columns):
-                inference_results.rename(columns={inference_results.columns[idx]: col}, inplace=True)
-              else: # assume later columns are copied from input
-                # assume columns with same name in input / output tables are the same
-                _, col_n = col.split('.')
-                for c in self.binding.input_columns:
-                  if c.split('.')[1] == col_n:
-                    inference_results[col] = inp_row[c]
-                    break
-
-            tables = self.get_tables(self.binding.output_columns)
-            for table in tables:
-              columns = [col for col in self.binding.output_columns if col.startswith(table + '.')]
-              tmp_df = inference_results[columns]
-              tmp_df = tmp_df.astype('object')
-
-              if self._dialect == 'mysql' or self._dialect == 'postgresql':
```

Comment:
Is it safe to delete the logic here?

Reply:
I think so.

Pandas dataframe .to_sql function internally handles all the databases supported by sqlalchemy.

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_sql.html#r689dfd12abe5-1

Reply:
Does it deal with the upserts and stuff?

Reply:
Nope, it doesn't support upsert. But according to our logic, this case never happens as we have cache

Reply:
Can you leave a comment here too?